src/app.js

```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+const compression = require('compression');
+const cookieParser = require('cookie-parser');
+const { join } = require('path');
+require('env2')('.env');
+const { router } = require('./routes/index');
+const serverError = require('./Error/serverError');
+
+const app = express();
+app.disabled('x-powered-by');
+app.set('port', process.env.PORT || 4000);
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(express.static(join(__dirname, '..', 'public')));
+app.use(compression());
+app.use(cookieParser());
+
+app.use(router);
+app.use(serverError);
```

Comment:
I noticed that you're handling server errors, but ignoring client errors like 404. It's important to handle these errors as well to provide a better user experience and improve the robustness of your application. Consider adding appropriate error handling for client-side errors to your code